tests/system_tests/test_launch/test_launch_public_api.py

```diff
@@ -45,6 +47,86 @@ def patched_push_to_run_queue_introspection(*args, **kwargs):
             )
 
 
+def test_upsert_run_queue(user):
```

Comment:
yikes ... this test is a bit of a hack, but if @KyleGoyette signed-off on this i won't block

Reply:
@kptkin Can you elaborate?

Reply:
I guess I'm specifically confused about whether you are saying "yikes" because of how I am avoiding the issue with running against `0.40.0` or something else?

Reply:
sorry few points:
1. yep avoiding the issue, but it's up to you
2. the way you mange the queue resource if you want to have some form of a context manager to clean up your resources after use would be good to make some fixture or alternatively create some mock to handle the clean up?

Reply:
@kptkin  Im under the impression we're expecting our system tests to run against the most recent version of the server. If we're testing against older versions then this test is expected to fail. If theres no tooling in place for checking the server version we should file that as a separate ticket and leave that as a TODO, not block a new public API impl on tooling.

Also, is this the first new endpoint added to the public API since we've adopted a new test infra? How have other folks adding APIs handled this?

Reply:
reading again the test, yeah i guess my biggest issue is with the first point... second is just style...

Reply:
1. i didn't block! as you can see by the green stamp!!!
2.  just asked to introduce this feature as that was a gap that was identified - what will happen in practice is that this test would never be updated (or it would be left for the sdk team to fix it when the missing tooling is introduced)
3. note that this tooling is a few hours of work - and even tho the SDK owns maintence of this repo, i think it is a better culture if we all try to contribute and make the development or the code base better in this company!
4. we added test both against the latest server and the earliest available exactly because we want dev to think about cases when the code would be used with older server and what would be the user experience in this case!